### PR TITLE
feat: redact anonymous attributes within feature events

### PR DESCRIPTION
**Pull request by keelerm84**

### CODE REVIEW

File: libs/internal/include/launchdarkly/context_filter.hpp [hunk 1]
```diff
@@ -34,6 +34,17 @@ class ContextFilter {
...
+    /**
+     * Filter the given context and produce a JSON value. If a provided context
+     * is anonymous, all attributes are redacted.
+     *
+     * Only call this method for valid contexts.
+     *
+     * @param context The context to redact.
+     * @return JSON suitable for an analytics event.
+     */
+    JsonValue filter_with_anonymous_redaction(Context const& context);
```
Comment (keelerm84) [inline libs/internal/include/launchdarkly/context_filter.hpp:46]: Wasn't sure what the casing convention was for this repo so don't hesitate to nitpick!

File: libs/internal/include/launchdarkly/context_filter.hpp [hunk 2]
```diff
@@ -48,6 +59,15 @@ class ContextFilter {
...
     };
 
+    /**
+     * Internal delegate function for filter and filter_with_anonymous_redaction.
+     *
+     * @param context The context to redact.
+     * @param redactAnonymous Whether to redact all attributes from anonymous contexts.
+     * @return JSON suitable for an analytics event.
+     */
+    JsonValue filter_(Context const& context, bool redactAnonymous);
```
Comment (keelerm84) [inline libs/internal/include/launchdarkly/context_filter.hpp:69]: Trailing `_` the preference? Or something like `internalFilter`?

Comment (cwaldren-ld) [reply libs/internal/include/launchdarkly/context_filter.hpp:69]: Should be able to call it just `filter`, as an overload

File: libs/internal/tests/context_filter_test.cpp [hunk 3]
```diff
@@ -173,3 +173,56 @@ TEST(ContextFilterTests, AllAttributesPrivateMultiContext) {
...
+            .Name("Bob")
+            .Anonymous(true)
+            .Set("email", "email.email@email")
+            .Set("array", {false, true, "bacon"})
+            .Set("object", Object{{"test", true}, {"second", false}})
+            .SetPrivate("isCat", false)
+            .AddPrivateAttribute("/object/test")
+            .Kind("org", "org-key")
+            .Set("isCat", true)
+            .Set("email", "cat@cat.cat")
```
Comment (cwaldren-ld) [inline libs/internal/tests/context_filter_test.cpp:215]: TIL that `cat` is a real TLD (Catalan): https://www.gandi.net/en-US/domain/tld/cat

**Review by cwaldren-ld — APPROVED** [review]

**Pull request merged**